alacritty/src/event.rs

```diff
@@ -1106,6 +1108,26 @@ impl input::Processor<EventProxy, ActionContext<'_, Notifier, EventProxy>> {
                     WindowEvent::KeyboardInput { input, is_synthetic: false, .. } => {
                         self.key_input(input);
                     },
+                    WindowEvent::IME(ime) => match ime {
+                        // TODO decide what to do with it, but in general we should stop update of
+                        // IME input methods.
+                        IME::Enabled => {},
+                        // The text is commited thus should be written to PTY.
+                        IME::Commit(text) => {
+                            self.ctx.display.ime_input = None;
+                            text.chars().for_each(|ch| self.received_char(ch));
+                        },
+                        // This text is being composed, thus should be shown as inline input to
+                        // the user.
+                        IME::Preedit(text, cursor_start, cursor_end) => {
+                            *self.ctx.dirty = true;
+                            self.ctx.display.ime_input =
+                                Some(IMEInput::new(text, cursor_start, cursor_end));
```

Comment:
Small comment on the winit API and something I've noticed designing some enum APIs myself: If the parameter here would be a struct instead of a tuple, we could just use the struct 1:1 for our own implementation. But since it's a tuple we need to take that tuple and convert it to our own type.

Generally this makes abstractions a lot more difficult, you can never just take `IME::Preedit(xxx)` and use the entire `xxx` nicely for future processing because then you'll always have to deal with tuple indexing. Using structs instead will automatically allow you to refer to the fields by their name, making the API a lot nicer to use in my experience.

Reply:
Yeah, I also thought that it'd be nice. Will refactor to send struct instead of tuple.